code/code/misc/parse.cc

```diff
@@ -2081,16 +2083,8 @@ void argument_interpreter(sstring argument, sstring &first_arg, sstring &second_
 
 bool is_number(const sstring &str)
 {
-  int look_at;
-
-  if (str.empty())
-    return (0);
-
-  for (look_at = 0; *(str.c_str() + look_at) != '\0'; look_at++) {
-    if ((*(str.c_str() + look_at) < '0') || (*(str.c_str() + look_at) > '9'))
-      return (0);
-  }
-  return (1);
+    boost::smatch what;  //Not really used here, since we're just doing go/no-go
+    return boost::regex_match(str, what, boost::regex("^[\\+-]?\\d+(\\.\\d+)?$"));
```

Comment:
This is the big one to watch, since it's used in a few places to validate input. Previously, it only returned true for text integers, but false for anything that had anything but a zero thru nine in it. So while it correctly invalidated words and letters and other garbage, it also came back as invalid for floating point numbers, negative numbers, and so on. This change fixes that. If the number is preceeded by a plus or minus, it will now be validated correctly. If it has a decimal, it must have at least one digit to the right, otherwise it will come back as invalid. And of course, no decimal point at all with/without signs will also work (signed/unsigned integer).